.github/workflows/cmake.yml

```diff
@@ -93,34 +92,42 @@ jobs:
       if: ${{ contains(matrix.os, 'ubuntu') }}
       run: |
         sudo apt update
-        sudo apt install openmpi-bin libopenmpi-dev ccache casacore-dev
-        pip install conan
+        sudo apt install openmpi-bin libopenmpi-dev ccache graphviz libeigen3-dev libspdlog-dev libtiff-dev libcfitsio-dev libbenchmark-dev libboost-all-dev libyaml-cpp-dev
+        git clone https://github.com/catchorg/Catch2.git -b v3.4.0
+        mkdir Catch2/build
+        cd Catch2/build
+        cmake .. -DBUILD_TESTING=OFF -DCMAKE_INSTALL_PREFIX=${{github.workspace}}/local
+        make -j$(nproc --ignore 1) install
+        cd -
+        wget --no-check-certificate --no-verbose http://www.fftw.org/fftw-3.3.10.tar.gz -O- | tar --no-same-owner -xz;
+        cd fftw-3.3.10
+        ./configure --prefix=${{github.workspace}}/local --enable-shared
+        make -j$(nproc --ignore 1) install CFLAGS=-fPIC
```

Comment:
There's `apt` packages for `catch2` and `fftw` on my Ubuntu 22.04. They are a few versions behind the latest. Was that why you didn't use them?

Reply:
The `catch2` package in `apt` does look ancient so we probably do want to git clone it. You could use the `checout` action to do it in a separate step, something like

```
- name: Checkout catch2 repo
  uses: actions/checkout@v4
  with:
    repository: catchorg/Catch2
    path: Catch2
    ref: v3.4.0
```

I'm kind of surprised no one has written an action for this